tests/firebird/firebird_integration_test.go

```diff
@@ -0,0 +1,283 @@
+	tests.RunMCPToolCallMethod(t, mcpInvokeParamWant, failInvocationWant)
+
+	templateParamTestConfig := tests.NewTemplateParameterTestConfig(
+		tests.WithIgnoreDdl(),
+		tests.WithIgnoreInsert(),
+		tests.WithSelect1Want(`[{"AGE":21,"ID":1,"NAME":"Alex"}]`),
+		tests.WithSelectAllWant(`[{"AGE":21,"ID":1,"NAME":"Alex"},{"AGE":100,"ID":2,"NAME":"Alice"}]`),
+		tests.WithReplaceNameFieldArray(`["NAME"]`),
+		tests.WithReplaceNameColFilter("NAME"),
+	)
+	tests.RunToolInvokeWithTemplateParameters(t, tableNameTemplateParam, templateParamTestConfig)
+}
+
+func setupFirebirdTable(t *testing.T, ctx context.Context, db *sql.DB, createStatements []string, insertStatement, tableName string, params []any) func(*testing.T) {
+	t.Helper()
+	err := db.PingContext(ctx)
+	if err != nil {
+		t.Fatalf("unable to connect to test database: %s", err)
+	}
+
+	for _, stmt := range createStatements {
+		_, err = db.ExecContext(ctx, stmt)
+		if err != nil {
+			t.Fatalf("unable to execute create statement for table %s: %s\nStatement: %s", tableName, err, stmt)
+		}
+	}
+
+	if insertStatement != "" && len(params) > 0 {
+		stmt, err := db.PrepareContext(ctx, insertStatement)
+		if err != nil {
+			t.Fatalf("unable to prepare insert statement: %v", err)
+		}
+		defer stmt.Close()
+
+		numPlaceholders := strings.Count(insertStatement, "?")
+		if numPlaceholders == 0 {
+			t.Fatalf("insert statement has no placeholders '?' but params were provided")
+		}
+		for i := 0; i < len(params); i += numPlaceholders {
+			end := i + numPlaceholders
+			if end > len(params) {
+				end = len(params)
+			}
+			batchParams := params[i:end]
+
+			_, err = stmt.ExecContext(ctx, batchParams...)
+			if err != nil {
+				t.Fatalf("unable to insert test data row with params %v: %v", batchParams, err)
+			}
+		}
+	}
+
+	return func(t *testing.T) {
+		t.Helper()
+		isNotFoundError := func(err error) bool {
+			if err == nil {
+				return false
+			}
+			errMsg := strings.ToLower(err.Error())
+			return strings.Contains(errMsg, "does not exist") || strings.Contains(errMsg, "not found") || strings.Contains(errMsg, "is not defined")
+		}
+
+		_, err := db.ExecContext(ctx, fmt.Sprintf("DROP TRIGGER BI_%s_ID;", tableName))
+		if err != nil && !isNotFoundError(err) {
+			t.Logf("Could not drop trigger (this might be ok): %s", err)
+		}
+		_, err = db.ExecContext(ctx, fmt.Sprintf("DROP TABLE %s;", tableName))
+		if err != nil && !isNotFoundError(err) {
+			t.Errorf("Teardown failed to drop table: %s", err)
+		}
+		_, err = db.ExecContext(ctx, fmt.Sprintf("DROP GENERATOR GEN_%s_ID;", tableName))
+		if err != nil && !isNotFoundError(err) {
+			t.Logf("Could not drop generator (this might be ok): %s", err)
+		}
+	}
+}
+
+func getFirebirdParamToolInfo(tableName string) ([]string, string, string, string, string, string, []any) {
+	createStatements := []string{
+		fmt.Sprintf("CREATE TABLE %s (ID INTEGER NOT NULL PRIMARY KEY, NAME VARCHAR(255));", tableName),
```

Comment:
```suggestion
		fmt.Sprintf("CREATE TABLE %s (id INTEGER NOT NULL PRIMARY KEY, name VARCHAR(255));", tableName),
```

Keep consistent with the other integration tests so we can utilizes some helper functions or default returns.